tests/unit/ml_handlers/test_langchain_embedding.py

```diff
@@ -0,0 +1,365 @@
+import os
+import time
+from unittest.mock import patch
+
+import pandas as pd
+import pytest
+from mindsdb_sql import parse_sql
+
+from ..executor_test_base import BaseExecutorTest
+
+
+class TestLangchainEmbedding(BaseExecutorTest):
+    def wait_predictor(self, project, name):
+        # wait
+        done = False
+        for _ in range(200):
+            ret = self.run_sql(f"select * from {project}.models where name='{name}'")
+            if not ret.empty:
+                if ret["STATUS"][0] == "complete":
+                    done = True
+                    break
+                elif ret["STATUS"][0] == "error":
+                    break
+            time.sleep(0.5)
+        if not done:
+            raise RuntimeError("predictor wasn't created")
+
+    def run_sql(self, sql):
+        ret = self.command_executor.execute_command(parse_sql(sql, dialect="mindsdb"))
+        assert ret.error_code is None
+        if ret.data is not None:
+            columns = [
+                col.alias if col.alias is not None else col.name for col in ret.columns
+            ]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_dummy_embedding(self, mock_handler):
+        self.run_sql("create database proj")
+        # create  the model
+        self.run_sql(
+            """
+            CREATE MODEL proj.test_dummy_embedding
+            PREDICT embeddings_output_column
+            USING
+                engine='langchain_embedding',
+                class = 'FakeEmbeddings',
+                size = 512,
+                input_columns = ['content']
+            """
+        )
+
+        self.wait_predictor("proj", "test_dummy_embedding")
+
+        # predictions
+        # one line
+        ret = self.run_sql(
+            """
+            SELECT * FROM proj.test_dummy_embedding
+            WHERE content='hello'
+            """
+        )
+        assert "content" in ret.columns
+        assert "embeddings_output_column" in ret.columns
+        # the embeddings should be a list of 512 floats
+        assert len(ret["embeddings_output_column"][0]) == 512
```

Comment:
Curious, how is this test actually passing? Shouldn't the `FakeEmbedding` class not be found and subsequently failing to produce an embedding?

Reply:
FakeEmbeddings is an available embedding model in langchain. It just spits out a random vector given a piece of text. https://python.langchain.com/docs/integrations/text_embedding/fake